plugins/module_utils/ansible_freeipa_module.py

```diff
@@ -190,6 +190,16 @@ def api_connect(context=None):
 
 def api_command(module, command, name, args):
     """Call ipa.Command."""
+    if module.check_mode:
+        ro_attr_names = ["conncheck", "defaults", "fetch_domains", "find",
+                         "find_orphans", "match", "request",
+                         "retrieve_internal", "show", "state", "status",
+                         "verify"]
+        try:
+            if api.Command[command].attr_name not in ro_attr_names:
+                return {}
+        except AttributeError:
+            return {}
```

Comment:
A similar approach needs to be done on ansible_freeia_module.api_command_no_name, or the ipaconfig and ipaconfig modules will not work with `--check`.

Maybe making the the check in another function and modifying the api_command*() function to `return api.Command[command](name, **args) if _is_check_disabled(module) else {}`.

Reply:
No longer needed as this code was removed.